Allow specifying kb-layout, course-title and -description (& cleanup)

hoijui opened the pull request:
Please tell me if you want only part of this, or want things changed, and I'll make separate PRs.

simgunz commented on ktouch-lesson-generator.py:
```diff
@@ -14,7 +14,10 @@
   -p, --plain-text                           Output the lessons in plain text instead of XML
   -w NUM, --word-wrap NUM                    Wrap lesson text at this length. [default: 60]
   -l NUM, --characters-per-lesson NUM        Number of characters in a lesson. [default: 2000]
+      --course-description STR               Sets the description for the course [default: ""]
```
For these three new parameters you cannot set the default here because that string e.g. `""` is used as default value and you get this as result:
```
<title>"KTouch-Generator-{shortid}"</title>
<description>""</description>
<keyboardLayout>""</keyboardLayout>
```
You can write the default as text not within square brackets and then process the default value that is `None`.

simgunz commented on ktouch-lesson-generator.py:
```diff
@@ -457,7 +469,8 @@ def formatLessonXML(currentChars, lessonText, lessonNumber='', lessonPrefix='Les
     if args['<dictionary>']:
         with open(args['<dictionary>']) as f:
-            # Consider only first column, strip newlines, strip hypnetion information from some dictionaries
+            # Consider only first column, strip newlines,
```
Do not reflow this comment, max line length for this project is 120

simgunz reviewed (changes requested):
Thanks for the contribution.

Please remove the commits where you reflow the README, I prefer the lines to be wrapped automatically by the github UI or the local markdown editor.

See my other comments regarding a few things to change.

hoijui commented:
Thank you for the review! :-)

I think I addressed all your concerns:

* No more reflowing in the README
* No more reflowing that one comment
* correct defaults for description, title and keyboard-layout
